proto/sentry_protos/sentry/v1alpha/taskworker.proto

```diff
@@ -0,0 +1,141 @@
+syntax = "proto3";
+
+package sentry_protos.sentry.v1alpha;
+
+import "google/protobuf/timestamp.proto";
+
+message RetryState {
+  // Current attempt number
+  int32 attempts = 1001;
+
+  // The classname or adapter type for the retry policy
+  string kind = 1002;
+
+  // After this attempt the task should be discarded
+  optional int32 discard_after_attempt = 1003;
+
+  // After this attempt the task should be put in the dead-letter-queue.
+  optional int32 deadletter_after_attempt = 1004;
+}
+
+// Task message that is stored in Kafka.
+// Once a task becomes a Pending task additional state is tracked.
+message WorkerTask {
+  // A GUID for the task. Used to update tasks
+  string task_id = 1;
+
+  // The name of the task. This name is resolved within the worker
+  string taskname = 2;
+
+  // An opaque parameter collection. Could be JSON or protobuf encoded
+  string parameters = 3;
+
+  // A map of headers for the task.
+  map<string, string> headers = 4;
+
+  // The timestamp a task was stored in Kafka
+  google.protobuf.Timestamp received_at = 5;
+
+  // The task deadline. After this time the task is expired and should not be executed.
+  optional google.protobuf.Timestamp deadline = 6;
+
+  // Retry state
+  RetryState retry_state = 7;
+}
+
+// Once a WorkerTask has been received by the task consumer it is 'upgraded'
+// into a PendingTask so that processing state can be tracked.
+// This proto might not be used as Pending tasks don't need to cross process boundaries.
+message PendingTask {
```

Comment:
Is the reason why we don't need to store information like topic name and partition because topic namespaces are assigned to specific worker groups? Therefore, this information is implicitly known?

Reply:
Yes, a few of the storage options we're considering could have storage partitioned by topic/partition. I thought having that information in PendingTasks would be redundant, but if we go with a storage solution that requires topic/partition we can add those fields.